src/pgduckdb_types.cpp

```diff
@@ -702,6 +850,31 @@ ConvertDecimal(const NumericVar &numeric) {
 	return (NumericIsNegative(numeric) ? -base_res : base_res);
 }
 
+int
+GetEnumPosition(Datum enum_oid) {
+	HeapTuple enum_tuple;
+	Form_pg_enum enum_entry;
+	int enum_position;
+
+	// Use SearchSysCache1 to fetch the enum entry from pg_enum using its OID
+	enum_tuple = SearchSysCache1(ENUMOID, enum_oid);
+
+	if (!HeapTupleIsValid(enum_tuple)) {
+		elog(ERROR, "could not find enum with OID %u", DatumGetObjectId(enum_oid));
+	}
+
+	// Get the pg_enum structure from the tuple
+	enum_entry = (Form_pg_enum)GETSTRUCT(enum_tuple);
+
+	// The enumsortorder field gives us the position of the enum value
+	enum_position = (int)enum_entry->enumsortorder;
```

Comment:
Actually, I misremembered the performance part of this comment. SearchSysCache is relatively efficient because it does not allocate, it's SearchSysCacheCopy that is particularly problematic performance wise.

Reply:
> I think it makes more sense to cache a mapping of postgres enum_oid to duckdb position in your custom enum type, so that we can get the position of an enum_oid without having to access postgres catalogs.

Instead of doing that, we could also simply iterate over the already existing `enum_member_oids` vec. Since the number of members is usually relatively small I think this will be fine performance wise, at least for a first version.

Reply:
I made the change 👍 
```c++
idx_t
PGDuckDBEnum::GetEnumPosition(Datum enum_member_oid_p, const duckdb::LogicalType &type) {
	auto enum_member_oid = DatumGetObjectId(enum_member_oid_p);

	auto &enum_type_info = type.AuxInfo()->Cast<duckdb::EnumTypeInfo>();
	auto dict_size = enum_type_info.GetDictSize();

	auto enum_member_oids = PGDuckDBEnum::GetMemberOids(type);
	auto oids_data = duckdb::FlatVector::GetData<uint32_t>(enum_member_oids);

	uint32_t *begin = oids_data;
	uint32_t *end = oids_data + dict_size;
	uint32_t *result = std::find(begin, end, enum_member_oid);

	if (result == end) {
		throw duckdb::InternalException("Could not find enum_member_oid: %d", enum_member_oid);
	}
	return (idx_t)(result - begin);
}
```